research-software-directory/RSD-as-a-service
# Migrate software index page

# Software index page (first itteraction)

Resolves #55 
Migrated most of the functionality of software page with desired improvements as described in #55.

How to test:

* remove old volumes: `docker-compose down --volumes`
* start new instance, migrate data and build: `docker compose up --build`
* open frontend at: http://localhost:3000
* navigate to Software page: http://localhost:3000/software
* type search term 'pharma': it should return 1 software
* try filtering on technologies using filter icon next to search bar
* when no match is found, basic `no content` message is shown (we plan to improve design later)

## Notes

* filtering on technologies is done by matching any of tags (logical OR). This is different from legacy version where software was require to have all selected values (logical AND). When selecting multiple technologies very few software meets the criteria.
* sorting using sort icon, next to filter icon, has ONLY one option enabled for now. Other options are not ready yet, as they require further DB migration of additional tables.
* `last updated message` in the card uses only updated_at information from the software table. See next point too
* `updated_at` date will be the date of data-migration. So, when testing it should be `1 hour ago` for all software (as you just created DB and migrated all data). We need to improve migration script to use original software values.
* current design might need futher improvements. Suggestions are welcome!

## Expected result 

![image](https://user-images.githubusercontent.com/9204081/146536040-c6c78d7f-b5ff-44d7-b64d-c8b4b97b6070.png)
